dynamiqs/integrators/apis/floquet.py

```diff
@@ -174,10 +174,13 @@ def _check_floquet_args(
     )
 
     # === check that the Hamiltonian is periodic with the supplied period
-    H = eqx.error_if(
-        H,
-        jnp.logical_not(jnp.isclose(H(0), H(T))),
-        'The Hamiltonian H is not periodic with the supplied period T.',
-    )
+    if not isinstance(H, CallableTimeArray):
+        # disable the check for CallableTimeArray as equinox cannot find an
+        # underlying array to attach the error message to.
+        H = eqx.error_if(
+            H,
+            jnp.logical_not(jnp.isclose(H(0.0)._data, H(T)._data)),  # noqa: SLF001
```

Comment:
This calls for a proper method to compare sparsedia qarrays with `jnp.isclose`, user might want to do this too. We can merge this as is and think about it.